plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -363,79 +348,177 @@ fu_mediatek_scaler_device_get_firmware_version(FuDevice *device, GError **error)
 				    G_LITTLE_ENDIAN,
 				    error))
 		return FALSE;
-	fw_version = mediatek_scaler_device_version_to_string(tmpversion);
-	fu_device_set_version(device, fw_version);
+	fu_device_set_version_raw(FU_DEVICE(self), version_raw);
 	return TRUE;
 }
 
 static gboolean
-fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
+fu_mediatek_scaler_device_open(FuDevice *device, GError **error)
 {
 	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
-	g_autoptr(FuUdevDevice) udev_parent = NULL;
-	g_autoptr(FuDeviceLocker) locker = NULL;
-	g_autofree gchar *vendor_id = NULL;
-	g_autofree gchar *vid = NULL;
-	g_autofree gchar *pid = NULL;
-	g_autofree gchar *hw_ver = NULL;
 
-	/* FuUdevDevice->probe */
-	if (!FU_DEVICE_CLASS(fu_mediatek_scaler_device_parent_class)->probe(device, error))
+	/* proxy */
+	if (!fu_device_open(FU_DEVICE(self->i2c_dev), error))
 		return FALSE;
 
-	/* determine the i2c_dev for dp aux dev */
-	if (!fu_mediatek_scaler_device_use_aux_dev(device, error))
+	/* set the target address */
+	if (!fu_mediatek_scaler_ensure_device_address(self,
+						      FU_DDC_I2C_ADDR_DISPLAY_DEVICE >> 1,
+						      error))
 		return FALSE;
 
-	/* open device */
-	locker = fu_device_locker_new(self->i2c_dev, error);
-	if (locker == NULL)
+	/* prioritize DDC/CI commands in display controller */
+	if (!fu_mediatek_scaler_device_set_ddc_priority(self, FU_DDCCI_PRIORITY_UP, error))
 		return FALSE;
 
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_mediatek_scaler_device_close(FuDevice *device, GError **error)
+{
+	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
+
 	/* set the target address */
 	if (!fu_mediatek_scaler_ensure_device_address(self,
 						      FU_DDC_I2C_ADDR_DISPLAY_DEVICE >> 1,
 						      error))
 		return FALSE;
 
-	/* prioritize DDC/CI commands in display controller */
-	if (!fu_mediatek_scaler_device_set_ddc_priority(device, FU_DDCCI_PRIORITY_UP, error))
+	/* reset DDC priority */
+	if (!fu_mediatek_scaler_device_set_ddc_priority(self, FU_DDCCI_PRIORITY_NORMAL, error))
+		return FALSE;
+
+	/* proxy */
+	if (!fu_device_close(FU_DEVICE(self->i2c_dev), error))
+		return FALSE;
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_mediatek_scaler_device_verify_controller_type(FuMediatekScalerDevice *self, GError **error)
+{
+	g_autoptr(GByteArray) st_req = fu_struct_ddc_cmd_new();
+	g_autoptr(GByteArray) st_res = NULL;
+	g_autoptr(GError) error_local = NULL;
+	guint32 controller_type = 0;
+
+	fu_struct_ddc_cmd_set_opcode(st_req, FU_DDC_OPCODE_GET_VCP);
+	fu_struct_ddc_cmd_set_vcp_code(st_req, FU_DDC_VCP_CODE_CONTROLLER_TYPE);
+	st_res = fu_mediatek_scaler_device_ddc_read(self, st_req, error);
+	if (st_res == NULL)
+		return FALSE;
+	if (!fu_memread_uint32_safe(st_res->data,
+				    st_res->len,
+				    st_res->len - 4,
+				    &controller_type,
+				    G_BIG_ENDIAN,
+				    error))
+		return FALSE;
+	fu_device_sleep(FU_DEVICE(self), FU_MEDIATEK_SCALER_DDC_MSG_DELAY_MS);
```

Comment:
presumably this delay should only be needed in the passing case (it can be moved to line 435)